### PR TITLE
DDF-2311 Fix security filters so they look up the context policy by t…

### PR DESCRIPTION
**Pull request by jrnorth**
#### What does this PR do?
-Fixes `WebSSOFilter` and `AuthorizationFilter` so they look up the context policy by the request URI rather than looking it up by the context path or the servlet path
-Delete unused `security-filter-delegate` module

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@codice/security
@bdeining

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@stustison
@pklinef

#### How should this be tested?
Verify that the default context realms, authentication types, required attributes, and whitelisted contexts in the Web Context Policy Manager work as expected.

Install the GeoWebCache app (which works well for testing since the GeoWebCache servlet has a non-empty context path and multiple endpoints). Create some new web context policies for GeoWebCache's endpoints (such as `/geowebcache/rest` or `/geowebcache/demo`) and verify that they work as expected.

#### What are the relevant tickets?
[DDF-2311](https://codice.atlassian.net/browse/DDF-2311)

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by pklinef** [comment]
👍 

@stustison Please take a look when you get a chance.

File: platform/security/filter/security-filter-delegate/src/main/java/org/codice/ddf/security/filter/delegate/DelegateServletFilter.java [hunk 1]
```diff
@@ -1,123 +0,0 @@
...
-import org.osgi.framework.ServiceReference;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-/**
- * {@link DelegateServletFilter} is meant to detect any Security ServletFilters
- * and call their {@link FilterChain}. If none are found it becomes a pass
- * through filter.
- */
-public class DelegateServletFilter implements Filter {
```
Comment (pklinef) [inline platform/security/filter/security-filter-delegate/src/main/java/org/codice/ddf/security/filter/delegate/DelegateServletFilter.java]: Why did you remove the `DelegatingServletFilter`?  This is part of a solution to get in front of servlet filters defined in web.xml.  For example, without this it is not possible to properly secure the Solr servlet filters.

👎  until this is resolved.

Comment (jrnorth) [inline platform/security/filter/security-filter-delegate/src/main/java/org/codice/ddf/security/filter/delegate/DelegateServletFilter.java]: I deleted the `platform/security/filter/security-filter-delegate` module because it's not being used anywhere.

Comment (pklinef) [inline platform/security/filter/security-filter-delegate/src/main/java/org/codice/ddf/security/filter/delegate/DelegateServletFilter.java]: Had me worried there.  Everything else looked good.

👍

**Comment by bdeining** [comment]
LGTM

**Comment by shaundmorris** [comment]
Looks like this should be back ported to 2.9.x as well, correct?

**Comment by jrnorth** [comment]
Yes, I'll make a PR on the `2.9.x` branch as well.

**Comment by stustison** [comment]
👍
